include/eos/core/LandmarkMapper.hpp

```diff
@@ -133,6 +133,45 @@ class LandmarkMapper
         }
     };
 
+
+    using LandmarkDefinitions = std::unordered_map<std::string, int>;
+    /**
+     * @param landmarks landmarks to index
+     * @return new vector with landmarks for which index can be defined
+     */
+    template<typename LandmarkType>
+    auto get_indexed_landmarks(const LandmarkCollection<LandmarkType>& landmarks,
+                               const LandmarkDefinitions* const landmark_definitions = nullptr) const {
```

Comment:
I don't think we should have a raw pointer here, if we can stay with value-semantics. I think `optional` would be the way to go here as well?

Reply:
Ok